lib/src/tools/format_tool.dart

```diff
@@ -58,6 +58,11 @@ class FormatTool extends DevTool {
   /// By default, nothing is excluded.
   List<Glob> exclude;
 
+  /// If true, pass whole directories with no exclusions to the formatter.
+  ///
+  /// This is helpful for large projects that may cause the formatter command to have too many arguments
+  bool collapseDirectories = false;
```

Comment:
With the updated implementation it seems like we could just do this by default and avoid putting this behind a flag at all.

Reply:
I will test consumers tomorrow.